v5/core/cr_authenticator.go

```diff
@@ -101,40 +96,87 @@ type ComputeResourceAuthenticator struct {
 
 const (
 	defaultCRTokenFilename = "/var/run/secrets/tokens/vault-token"
-	defaultImdsEndpoint    = "http://169.254.169.254"
-	imdsVersionDate        = "2021-07-15"
-	imdsMetadataFlavor     = "ibm"
-	iamGrantTypeCRToken    = "urn:ibm:params:oauth:grant-type:cr-token" // #nosec G101
-	crtokenLifetime        = 300
+	iamGrantTypeCRToken = "urn:ibm:params:oauth:grant-type:cr-token" // #nosec G101
 )
 
 var craRequestTokenMutex sync.Mutex
 
-// NewComputeResourceAuthenticator constructs a new ComputeResourceAuthenticator instance with the supplied values and
-// invokes the ComputeResourceAuthenticator's Validate() method.
-func NewComputeResourceAuthenticator(crtokenFilename string, instanceMetadataServiceURL, iamProfileName string, iamProfileID string,
-	url string, clientID string, clientSecret string, disableSSLVerification bool, scope string,
-	headers map[string]string) (*ComputeResourceAuthenticator, error) {
-	authenticator := &ComputeResourceAuthenticator{
-		CRTokenFilename:            crtokenFilename,
-		InstanceMetadataServiceURL: instanceMetadataServiceURL,
-		IAMProfileName:             iamProfileName,
-		IAMProfileID:               iamProfileID,
-		URL:                        url,
-		ClientID:                   clientID,
-		ClientSecret:               clientSecret,
-		DisableSSLVerification:     disableSSLVerification,
-		Scope:                      scope,
-		Headers:                    headers,
-	}
+// ComputeResourceAuthenticatorBuilder is used to construct an instance of the ComputeResourceAuthenticator
+type ComputeResourceAuthenticatorBuilder struct {
+	ComputeResourceAuthenticator
```

Comment:
Note that the builder really just embeds the actual authenticator struct to hold all the properties.   Makes it pretty convenient in the various setter and Build() methods below.